docs/release.md

```diff
@@ -0,0 +1,48 @@
+# Release process
+
+This project follows [semantic versioning](https://semver.org/). The following
+documentation will refer to `X.Y.Z` as _major_, _minor_ and _patch_ version.
+
+## Development between releases
+
+- Every substantial pull request should add an entry to the `[unreleased]`
+  section of the corresponding crate `CHANGELOG.md` file. See
+  [#1698](https://github.com/libp2p/rust-libp2p/pull/1698/files) as an example.
+  
+  In case there is no `[unreleased]` section yet, create one with an increased
+  major, minor or patch version depending on your change. In addition update the
+  corresponding entry of the crate in the root level `Cargo.toml`.
+
+
+## Releasing one or more crates
+
+1. Set the release date for each crate to be released in the respective
+   `CHANGELOG.md` by replacing `# X.Y.Z [unreleased]` with
+   `# X.Y.Z [yyyy-mm-dd]` and create a pull request against the rust-libp2p
```

Comment:
I wonder why we insist on putting those dates in there? The version corresponds to a tag which maps to a commit which has a date. How often do people look at the date in a changelog?

Reply:
I am fine either way. I just want to be consistent across all `github.com/libp2p/rust-libp2p` crates. @romanb thoughts?

Reply:
I don't care about the dates in the changelogs much. That tradition was started [here](https://github.com/libp2p/rust-libp2p/pull/1614#discussion_r442206372).